src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/UnitTests/Controllers/HomeControllerTests.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using DevAdventCalendarCompetition.Controllers;
+using DevAdventCalendarCompetition.Models.Test;
+using DevAdventCalendarCompetition.Repository.Models;
+using DevAdventCalendarCompetition.Services.Interfaces;
+using DevAdventCalendarCompetition.Services.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Xunit;
+using static DevAdventCalendarCompetition.Tests.TestHelper;
+
+namespace DevAdventCalendarCompetition.Tests.UnitTests.Controllers
+{
+    public class HomeControllerTests
+    {
+        private readonly Mock<IHomeService> _homeServiceMock;
+
+        public HomeControllerTests()
+        {
+            this._homeServiceMock = new Mock<IHomeService>();
+        }
+
+        [Fact]
+        public void Index_CurrentTestDtoIsNull_ReturnsEmptyViewResult()
+        {
+            // Arrange
+            this._homeServiceMock.Setup(x => x.GetCurrentTests()).Returns((List<TestDto>)null);
+            using var controller = new HomeController(this._homeServiceMock.Object);
+
+            // Act
+            var result = controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.Model);
+        }
+
+        [Fact]
+        public void Index_UserIsNull_ReturnsAViewResutWithListOfTestDto()
```

Comment:
typo: `Result`

Reply:
I think this logic is wrong (logic in Controller). We should show tests only for logged user

Reply:
Typo corrected. Regarding logic - could be resolved as Mateusz did related changes with other PR.